### ISSUE TITLE
The base code running stuck,when using ddp strategy with multi-GPUS

### ISSUE THREAD

**Issue by ForeverMining** [facebookresearch/silk#30]
When I tested the pre-trained model, the results compared with superpoint did not meet expectations. I want to reproduce the effect in the paper. When using single-machine multi-card training, I encountered a deadlock problem in ddp mode. How should I modify the code to solve this problem. I am using pytorch lightning for the first time and have little experience。The software version is recommended in README & requirements.txt

**Comment by gleize** [comment]
Hi @ForeverMining,

> When I tested the pre-trained model, the results compared with superpoint did not meet expectations.

Can you share the details ? I'm curious to know under what circumstances it didn't meet expectations.

> When using single-machine multi-card training, I encountered a deadlock problem in ddp mode. How should I modify the code to solve this problem.

The DDP deadlock is a known problem with the PyTorch Lightning library. I had ran into similar deadlocks when using an older version of PyTorch Lightning. Once I updated to the version mentioned in the `requirements.txt` file, I haven't had a single deadlock and was able to run all the experiments presented in the paper.

There are two things you can do.
1. Make sure the PyTorch and PyTorch Lightning match exactly what we recommend using. If you didn't follow our exact setup script, you might end up with a slightly different version that causes the issue.
2. If you have the exact same version, the best thing to try is to update PyTorch Lightning and try again. This might require small changes in the codebase if the PyTorch Lightning library changed its API.

**Comment by ForeverMining** [comment]
Thanks  for your reply! This problem is indeed brought about by the framework. I've tried different versions, and solutions from the framework community, all with this problem. But there is no such problem when using dual-card training, so I changed the online data enhancement method, and the speed of part of the training can be accelerated by adjusting the CPU and GPU usage. This effectively solves the problem of efficiency.
